joomla/joomla-cms
# Bug #27080 fix: Incorect id generated for input fields in repetable s…

Bug #27080

Summary of Changes
For repetable subform if the name of any field in the subform contains hyphen (-) then the id of the clone is not generated correctly.

If we click on (+) button to add the clone of the subform then the id of the field with hyphen (-) in name do not contain correct number of the clone, instead X is shown in the id.

Testing Instructions
Check the value of id attribute of input element in repetable subform after clicking the plus (+) button to add new clone of sub-form with a field having name hyphen (-) in it.

E.g If we have repetable subform field named as 'office-address' then the id of the elements in the cloned set should be like this 'jform_office_address__office_address1__address' but it is generated as 'jform_office_address__office_addressX__address'

Expected result
The value of id attribute of input element should contain clone number and not X

Actual result
The clone number for the field is not getting replaced with X

Documentation Changes Required
no change required in documentation
